packages/twenty-server/src/core/user-workspace/user-workspace.entity.ts

```diff
@@ -22,15 +23,22 @@ export class UserWorkspace {
   @PrimaryGeneratedColumn('uuid')
   id: string;
 
+  @Field()
   @JoinColumn({ name: 'userId' })
   user: User;
 
+  @Field({ nullable: false })
   @Column()
   userId: string;
 
+  @Field(() => Workspace)
+  @ManyToOne(() => Workspace, (workspace) => workspace.users, {
```

Comment:
I would say that it should be a ManyToMany relation as users can have multiple workspaces, and workspaces can have multiple users

Reply:
I think it makes sense to keep ManyToOne here since its a user-workspace mapping entity.  We can add a ManyToMany relation field on respective `User` and `Workspace` entities (I have commented out the code for now because I can't get it to work)

I saw a few examples of ManyToMany on TypeOrm docs which resembles this condition -https://github.com/typeorm/typeorm/blob/master/docs/relations.md#jointable-options https://orkhan.gitbook.io/typeorm/docs/many-to-many-relations#bi-directional-relations

Let me know what you think :)

Reply:
yes, you are right